store/store_action.go

```diff
@@ -564,3 +579,47 @@ func findRev(needle snap.Revision, haystack []snap.Revision) bool {
 	}
 	return false
 }
+
+func reportFetchAssertionsError(res *snapActionResult, assertq AssertionQuery) error {
+	// prefer to report the most unexpected error
+	e := -1
+	errl := res.ErrorList
+	carryingRef := func(i int) bool {
+		aType := asserts.Type(errl[i].Type)
+		return aType != nil && len(errl[i].PrimaryKey) == len(aType.PrimaryKey)
+	}
+	for i, ent := range errl {
+		withRef := carryingRef(i)
+		if withRef && ent.Code == "not-found" {
+			if e == -1 {
+				e = i
+			}
+		} else if withRef {
+			if e == -1 || errl[e].Code == "not-found" {
+				e = i
+			}
+		} else {
+			if e == -1 || carryingRef(e) {
```

Comment:
I understand the first part of this conditional (if e is not set yet we take the error) but I'm confused by "or" part; perhaps the comment I requested above will make it clear. Alternatiely, could you add a short comment here?